src/main/java/de/zalando/aruha/nakadi/config/RepositoriesConfig.java

```diff
@@ -36,15 +38,25 @@
     private ZookeeperConfig zookeeperConfig;
 
     @Bean
-    public EventTypeRepository eventTypeRepository() throws Exception {
-        final EventTypeRepository dbRepo = new EventTypeDbRepository(jdbcTemplate, jsonConfig.jacksonObjectMapper());
-
+    public EventTypeCache eventTypeCache() throws Exception {
         final String connectString = zookeeperConfig.zooKeeperHolder().get().getZookeeperClient().getCurrentConnectionString();
         final RetryPolicy retryPolicy = new ExponentialBackoffRetry(1000, 3);
         final CuratorFramework client = CuratorFrameworkFactory.newClient(connectString, retryPolicy);
 
         client.start();
 
-        return new EventTypeCachedRepository(dbRepo, new EventTypeCache(dbRepo, client));
+        EventBodyMustRespectSchema strategy = new EventBodyMustRespectSchema();
+        ValidationStrategy.register(EventBodyMustRespectSchema.NAME, strategy);
+
+        return new EventTypeCache(dbRepo(), client);
+    }
+
+    @Bean
+    public EventTypeRepository eventTypeRepository() throws Exception {
+        return new EventTypeCachedRepository(dbRepo(), eventTypeCache());
+    }
+
+    private EventTypeRepository dbRepo() {
```

Comment:
@tmuehl I'm trying to solve this intertwined dependencies issues. To explain why it's the way it is:

1. `EventTypeCachedRepository` needs an `EventTypeCache` because it needs to notify the cache when something changes in the repository. E.g. when someone updates an event type, we need to notify zookeeper that it changed. Talking to zookeeper is responsibility of `EventTypeCache`.

2. `EventTypeCache` needs an `EventTypeDbRepository` because it loads `EventType`s from database on cache miss.

3. `EventTypeCachedRepository` needs one `EventTypeDbRepository` because it needs to persist changes in the Db.